### PR TITLE
Blob versioning

### PR DESCRIPTION
**Pull request by mohsha-msft**

### CODE REVIEW

File: azblob/sas_service.go [hunk 1]
```diff
@@ -44,6 +44,14 @@ func (v BlobSASSignatureValues) NewSASQueryParameters(credential StorageAccountC
 			return SASQueryParameters{}, err
 		}
 		v.Permissions = perms.String()
+	} else if v.Version != null && v.Version != "" {
+		resource = "bv"
+		//Make sure the permission characters are in the correct order
+		perms := &BlobSASPermissions{}
+		if err := perms.Parse(v.Permissions); err != nil {
+			return SASQueryParameters{}, err
+		}
```
Comment (zezha-msft) [inline azblob/sas_service.go:53]: testing needed

File: azblob/url_blob.go [hunk 2]
```diff
@@ -108,16 +120,22 @@ func (b BlobURL) Undelete(ctx context.Context) (*BlobUndeleteResponse, error) {
 // does not update the blob's ETag.
 // For detailed information about block blob level tiering see https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers.
 func (b BlobURL) SetTier(ctx context.Context, tier AccessTierType, lac LeaseAccessConditions) (*BlobSetTierResponse, error) {
-	return b.blobClient.SetTier(ctx, tier, nil, RehydratePriorityNone, nil, lac.pointers())
+	return b.blobClient.SetTier(ctx, tier, nil,
+		nil, // Blob versioning
```
Comment (zezha-msft) [inline azblob/url_blob.go:124]: >// Blob versioning [](start = 7, length = 18)

please change these comments to clarify that the blob version is already included in the URL

**Comment by zezha-msft** [comment]
We need some basic testing for the version functionality, the same coverage as the Python PR would be nice.

File: azblob/url_blob.go (hunk 2 above)
Comment (mohsha-msft) [reply azblob/url_blob.go:131]: Done

**Pull request closed without merging**